### ISSUE TITLE
Why does xUnit deal with ExecutionContext - isn't it handled in the Async method builders/async infrastructure?

### ISSUE THREAD

**Issue by SpiritBob** [xunit/xunit#2355]
I'm curious to learn why there are classes that help deal with ExecutionContext, and manual setting/capturing of it, when I believe those things are part of the TPL flow - is it because we are not using the TPL?

Resources: https://devblogs.microsoft.com/pfxteam/executioncontext-vs-synchronizationcontext/

**Comment by bradwilson** [comment]
TL;DR: We need it to implement SynchronizationContext.Post. https://github.com/xunit/xunit/blob/a8614d34999c889e1c8014f679de95d20eae1304/src/xunit.execution/Sdk/MaxConcurrencySyncContext.cs#L59-L72

Longer answer:

We are not using TPL here (specifically, we're not using async methods here). We are implementing a sync context in order to limit the number of threads that are allowed to run concurrently, and so we need to capture the execution context and run on it when we're called to post a callback onto the same sync context/execution context.

While execution context may feel like an implementation detail for most developers, it's precisely the implementation detail that we need access to in order to _implement_ a sync context (we're not just using them, we're implementing them).

**Comment by bradwilson** [comment]
Closing this issue. If you want to have further discussion on the topic, please open it as a [discussion item](https://github.com/xunit/xunit/discussions?discussions_q=is%3Aunanswered) instead of an issue. Thanks!
